FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -7,36 +7,75 @@ import CoreData
 public final class CoreDataFeedStore: FeedStore {
 	private static let modelName = "FeedStore"
 	private static let model = NSManagedObjectModel(name: modelName, in: Bundle(for: CoreDataFeedStore.self))
-
+	
 	private let container: NSPersistentContainer
 	private let context: NSManagedObjectContext
-
+	
 	struct ModelNotFound: Error {
 		let modelName: String
 	}
-
+	
 	public init(storeURL: URL) throws {
 		guard let model = CoreDataFeedStore.model else {
 			throw ModelNotFound(modelName: CoreDataFeedStore.modelName)
 		}
-
+		
 		container = try NSPersistentContainer.load(
 			name: CoreDataFeedStore.modelName,
 			model: model,
 			url: storeURL
 		)
 		context = container.newBackgroundContext()
 	}
-
+	
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		perform { context in
+			do {
+				if let coreDataFeed = try CoreDataFeed.fetch(in: context) {
+					completion(.found(feed: coreDataFeed.localFeed, timestamp: coreDataFeed.timestamp))
+				} else {
+					completion(.empty)
+				}
+			} catch {
+				completion(.failure(error))
+			}
+		}
 	}
-
+	
 	public func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
-		fatalError("Must be implemented")
+		perform { context in
+			do {
+				let newCache = try CoreDataFeed.newInstance(in: context)
+				
+				let value = CoreDataFeed.managedFeedSet(from: feed, in: context)
+				
+				newCache.coreDataFeedImages = value
```

Comment:
To follow up on the comment on `coreDataFeedImages` please rename it to `images`

```suggestion
				newCache.images = value
```

Reply:
Done